src/components/shared/swipeable-modal/SwipeableModal.tsx

```diff
@@ -215,26 +360,65 @@ class SwipeableModal extends React.Component<
       position: positionTarget,
       targets: animateObject,
       update: () => {
-        this.setState({
+        setModalState({
+          ...modalState,
           lastPosition: animateObject.position,
-          opacity: 1 - this.swipeRatio(animateObject.position, swipableWidth),
+          opacity: 1 - swipeRatio(animateObject.position, swipableWidth),
           swiping: false,
           translateX: `${animateObject.position}px`,
         });
       },
     });
-  }
-
-  private swipeRatio(position: number, width: number) {
-    const initialPosition = this.initialDivPosition;
-    const diff = position - initialPosition;
+  };
 
-    return diff / width;
-  }
+  const onAction = () => {
+    close();
+  };
 
-  private onAction() {
-    this.close();
-  }
+  return (
+    <>
+      {!open && <></>}
```

Comment:
Why not just an early return like it used to be?

Reply:
Implemented